### PR TITLE
Partial sparks boilerplate reduction

### PR DESCRIPTION
**Pull request by coiax**
- Lots of objects are making `/datum/effect_system/spark_spread`, setting
them up, firing them, and then not qdeling them afterwards.
- Makes a `do_sparks` global proc that takes the same arguments as `set_up`
and then calls `start` and then `qdel`
- Switches a bunch of things to use this proc
- Makes emitters keep their own spark_spread datum around, since they
spark so damn much.
- Also makes some things use timers and ports some things to Initialize.

### CODE REVIEW

**Comment by kevinz000** [comment]
don't forget portals they do that too lol.

**Comment by RemieRichards** [comment]
@coiax smoke next, has the same thing as sparks in a lot of cases (make effect system, set it up, set it off, never remember to delete it)

in its own PR though.

**Comment by coiax** [comment]
@RemieRichards I don't understand why this isn't compiling

File: code/modules/projectiles/projectile/bullets.dm [hunk 1]
```diff
@@ -49,9 +49,7 @@
 	..()
 
 /obj/item/projectile/bullet/pellet/weak/on_range()
- 	var/datum/effect_system/spark_spread/sparks = new /datum/effect_system/spark_spread
- 	sparks.set_up(1, 1, src)
- 	sparks.start()
+	do_sparks(1, TRUE, src)
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: Check the whitespace here

**Pull request merged**